apps/meteor/app/livechat/server/api/v1/contact.ts

```diff
@@ -133,10 +133,17 @@ API.v1.addRoute(
 	{ authRequired: true, permissionsRequired: ['view-livechat-contact'], validateParams: isGETOmnichannelContactsProps },
 	{
 		async get() {
-			if (!isSingleContactEnabled()) {
-				return API.v1.unauthorized();
+			const { contactId, visitor } = this.queryParams;
+
+			if (!contactId && !visitor) {
+				return API.v1.notFound();
+			}
+
+			const contact = await (contactId ? LivechatContacts.findOneById(contactId) : getContactByChannel(visitor));
```

Comment:
we have some libs, wonder if this would be better inside a lib and we do the decision inside there.

Reply:
Gonna flag this one as resolved with no change.